### PR TITLE
Idp 781 update last login after mfa

### PR DESCRIPTION
**Pull request by hobbitronics**
[IDP-781](https://itse.youtrack.cloud/issue/IDP-781)

### Fixed
- update last_login_utc after mfa verification

---

### Feature PR Checklist
- [ ] Documentation (README, local.env.dist, api.raml, etc.)
- [ ] Tests created or updated
- [ ] Run `make composershow`
- [ ] Run `make psr2`

### CODE REVIEW

File: application/frontend/controllers/MfaController.php [hunk 1]
```diff
@@ -115,6 +115,14 @@ public function actionVerify(int $id, string $type = "")
             throw new BadRequestHttpException();
         }
         $mfa->loadData($rpOrigin);
+        $user->last_login_utc = MySqlDateTime::now();
```
Comment (briskt) [inline application/frontend/controllers/MfaController.php:118]: I think this controller action is also used to verify a new webauthn, but I could be wrong.

**Review by briskt — COMMENTED** [review]
It's going to be more tricky than this, because the same endpoint is used to verify a new webauthn, and we probably don't want to set the date in that case. Though I suppose it wouldn't be terrible since they would have had to login not long prior to that time in order to add the webauthn.

File: application/frontend/controllers/MfaController.php (hunk 1 above)
Comment (hobbitronics) [reply application/frontend/controllers/MfaController.php:118]: It looks that way from the tests. Any ideas on how to only update it when logging in? I was wondering about only updating if the previous one is older than the token lifespan.

**Comment by hobbitronics** [comment]
> It's going to be more tricky than this, because the same endpoint is used to verify a new webauthn, and we probably don't want to set the date in that case. Though I suppose it wouldn't be terrible since they would have had to login not long prior to that time in order to add the webauthn.

We would also need to update the last_login_utc when mfa isn't required (30 days or so after using it). That data doesn't seem to be available here unless its stored in the user. I am guessing it is stored as a cookie or in a session by simpleSaml on the client. I am starting to think this is the tricky part.

**Pull request closed without merging**